app/models/kuroko2/job_definition.rb

```diff
@@ -100,6 +101,18 @@ def create_instance(script: nil, launched_by:, token: nil )
     job_instances.create!(script: script, log_message: message)
   end
 
+  def save_and_record_revision(edited_user: nil)
```

Comment:
> This should be executed in the same transaction with @definition.save, I think.
Since revision requires user_id, running record_revision in after_save hook looks bad idea to me.

I think @eagletmt means not only make it in one method, but execute in the real `transaction`, doesn't it?

Reply:
I fixed in 10663408eccc18bbe467afc0f37b58bc24775fbe.